### DISCUSSION TITLE
I cannot access my Synology at http://172.15.0.1?

### DISCUSSION THREAD

**Discussion by cnlibo** [wg-easy/wg-easy#983] [Q&A] [answered]
I built wg-easy at my Synolody's docker with the following enviroment config:

> --name=wg-easy
-e WG_HOST= mydomain
-e PASSWORD= 1234&1607
-e WG_DEVICE= ovs_eth0
-e WG_DEFAULT_ADDRESS= 172.15.0.x
-e WG_DEFAULT_DNS= 114.114.114.114
-e WG_ALLOWED_IPS= 172.15.0.0/24
-e WG_POST_UP= iptables -A FORWARD -i %i -j ACCEPT; iptables -t nat -A POSTROUTING -o ovs_eth0 -j MASQUERADE
-e WG_POST_DOWN= iptables -D FORWARD -i %i -j ACCEPT; iptables -t nat -D POSTROUTING -o OVS_eth0 -j MASQUERADE
-v ~/.wg-easy:/etc/wireguard
-p 51820:51821/udp
-p 51821:51820/tcp
--cap-add=NET_ADMIN
--cap-add=SYS_MODULE
--sysctl="net.ipv4.conf.all.src_valid_mark=1"
--sysctl="net.ipv4.ip_forward=1"
--restart unless-stopped
ghcr.io/wg-easy/wg-easy

I created 3 peers staring from 172.15.0.1 to 172.15.0.3.
Peers connect each other perfectly, but when I tried to access my Synoloy at http://172.15.0.1:5000/, nothing happens.

When I SSH synology and imput "ifconfig", I got the following:

> docker0 Link encap:Ethernet HWaddr 02:42:8D:4A:74:76
inet addr:172.17.0.1 Bcast:172.17.255.255 Mask:255.255.0.0
inet6 addr: fe80::42:8dff:fe4a:7476/64 Scope:Link
UP BROADCAST RUNNING MULTICAST MTU:1500 Metric:1
RX packets:8039 errors:0 dropped:0 overruns:0 frame:0
TX packets:10211 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:1000
RX bytes:4535091 (4.3 MiB) TX bytes:4374583 (4.1 MiB)
> 
> docker99d Link encap:Ethernet HWaddr F6:A0:4E:37:E4:CB
inet6 addr: fe80::f4a0:4eff:fe37:e4cb/64 Scope:Link
UP BROADCAST RUNNING MULTICAST MTU:1500 Metric:1
RX packets:373 errors:0 dropped:0 overruns:0 frame:0
TX packets:529 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:0
RX bytes:125076 (122.1 KiB) TX bytes:170796 (166.7 KiB)
> 
> eth0 Link encap:Ethernet HWaddr 02:11:32:24:E7:DB
UP BROADCAST RUNNING SLAVE MULTICAST MTU:1500 Metric:1
RX packets:33712921 errors:0 dropped:0 overruns:0 frame:0
TX packets:64438256 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:1000
RX bytes:3420641838 (3.1 GiB) TX bytes:75180731232 (70.0 GiB)
> 
> lo Link encap:Local Loopback
inet addr:127.0.0.1 Mask:255.0.0.0
inet6 addr: ::1/128 Scope:Host
UP LOOPBACK RUNNING MTU:65536 Metric:1
RX packets:114837 errors:0 dropped:0 overruns:0 frame:0
TX packets:114837 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:1
RX bytes:13548772 (12.9 MiB) TX bytes:13548772 (12.9 MiB)
> 
> ovs_eth0 Link encap:Ethernet HWaddr 02:11:32:24:E7:DB
inet addr:192.168.1.100 Bcast:192.168.1.255 Mask:255.255.255.0
inet6 addr: fe80::11:ff::e7db/64 Scope:Link
inet6 addr: 240e:b65:50:11:32ff4:e7db/64 Scope:Global
UP BROADCAST RUNNING MULTICAST MTU:1500 Metric:1
RX packets:33516005 errors:0 dropped:0 overruns:0 frame:0
TX packets:54782100 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:1
RX bytes:2938450202 (2.7 GiB) TX bytes:74562243228 (69.4 GiB)

Can anybody help me? Thanks in advance!

**Comment by eandersons** [comment]
`WG_ALLOWED_IPS=172.15.0.0/24` means that only requests to IPs within range from 172.15.0.0 to 172.15.0.255 will be routed through WireGuard Easy.
You could try to use `WG_ALLOWED_IPS=172.15.0.0/24, 172.17.0.1/32` to see if that allows you to reach your Synology NAS.

**Comment by cnlibo** [comment]
Thanks for your quick reply.
I've tried but failed. 

172.17.0.1 is Docker0's address, which does not change no matter I start the "wg-easy" from docker or not, so I don't think it should be the Synology's IP address.

When I use wireguard command line, after typing "sudo wg-quick up wg0", there will be a virtual ethernet called "wg0" whose IP is 172.15.0.1, thus I can access the synology at http://172.15.0.1:5000

But after I turned to wg-easy, synology cannot be visited this way.
But when I pinged 172.15.0.1, It's replied as follows:

> 正在 Ping 172.15.0.1 具有 32 字节的数据:
来自 172.15.0.1 的回复: 字节=32 时间=2ms TTL=64
来自 172.15.0.1 的回复: 字节=32 时间=3ms TTL=64
来自 172.15.0.1 的回复: 字节=32 时间=2ms TTL=64
来自 172.15.0.1 的回复: 字节=32 时间=2ms TTL=64

**Comment by pheiduck** [comment]
Then its maybe a Port issue on your NAS System, you have to make it open at least for the IP Range of your internal and VPN Network.

**Comment by cnlibo** [comment]
Thanks Pheiduck！
I'm not sure as port 5000 is green-lighted in my firewall and I could accesss my Synology before I install "wg-easy" by the way of wireguard command line at http://172.15.0.1:5000 (5000 is the default port of synology).

**Comment by cnlibo** [comment, marked as answer]
### **Solved finally!**

I find the reason: **wg-easy** does not setup client configuration for Synololy, thus you'll have to manually add a client configuration at Synology. The complete steps are as follows:
1) download and install "wireguard" from "community" at Package Center in Synology, but NOT start it for now;
2) SSH Synolody and run sudo -i;
3) mkdir /etc/wireguard then cd /etc/wireguard;
4) vi wg0.conf as follows:
> 
	[Interface]
	PrivateKey = yCP5f1zDkB/MmeOXEhIc8H0=
	Address = 172.15.0.7/24
>
	[Peer]
	PublicKey = bXoP4KVhwNITk=
	PresharedKey = 2KSdTSh1YT34b2uQVA=
	AllowedIPs = 172.15.0.0/24
	PersistentKeepalive = 25
	Endpoint = yourdomain:51820
5) start Wiregard now;
6) sudo wg-quick up wg0;
7) ifconfig;
    Now you will be able to find a virtual ethernet called "wg0 00";
8) sudo wg-autostart enable wg0

Now it's done and you can access the Synology at http://172.15.0.7:5000 now.
